lib/request-handler.js

```diff
@@ -143,19 +142,19 @@ RequestHandler.prototype.log = utils.log;
 /**
  * Gets an available connection and sends the request
  * @param request
- * @param {QueryOptions} options
+ * @param {QueryOptions|null} options
  * @param {Function} callback
  */
 RequestHandler.prototype.send = function (request, options, callback) {
   if (this.request === null) {
-    //Set the first time
-    if (options && options.captureStackTrace) {
+    options = options || utils.emptyObject;
+    if (options.captureStackTrace) {
       //noinspection JSUnresolvedFunction
       Error.captureStackTrace(this.stackContainer);
     }
-    if (options && options.retry) {
-      this.retryPolicy = options.retry;
-    }
+    var profile = options.executionProfile || this.profileManager.getDefault();
```

Comment:
> In createQueryOptions, if there is no matching profile found, options.executionProfile gets set to utils.emptyObject 

Now that I think about it,  instead of setting options.executionProfile to utils.emptyObject, why not set it to profileManager.getDefault() there?  I think it would solve this problem as a side effect.